### PR TITLE
make import repo generic

### PR DESCRIPTION
**Pull request by malaffoon**
I will have three (?) PRs to make what is currently `exam-service` into a more generic `import-service`. This first set of changes is going from the bottom-up making the repo and service layers more generic. @rcordeau, this is so you can use the same service for package import.

### CODE REVIEW

File: common/src/main/java/org/opentestsystem/rdw/ingest/model/ImportContent.java [hunk 1]
```diff
@@ -5,7 +5,8 @@
  */
 public enum ImportContent {
     PROBE(0),
-    EXAM(1);
+    EXAM(1),
+    PACKAGE(2);
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/ingest/model/ImportContent.java]: There is a change in RDW_Schema to go with this.

File: common/src/main/java/org/opentestsystem/rdw/ingest/model/RdwImportQuery.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package org.opentestsystem.rdw.ingest.model;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+/**
+ * Container for query parameters.
+ */
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/ingest/model/RdwImportQuery.java]: Again, this isn't strictly speaking necessary for this change; i could've left all the findBy() parameters as individual values. But using a query object is a bit cleaner.

File: exam-service/src/main/java/org/opentestsystem/rdw/ingest/repository/RdwImportRepositoryImpl.java [hunk 3]
```diff
@@ -140,6 +121,32 @@ public void delete(final long id) {
     }
 
     @Override
+    public List<RdwImport> findBy(final RdwImportQuery query) {
+        // TODO - better way to do this using application.sql.yml?
```
Comment (malaffoon) [inline exam-service/src/main/java/org/opentestsystem/rdw/ingest/repository/RdwImportRepositoryImpl.java]: @agorina thoughts?

Comment (agorina) [inline exam-service/src/main/java/org/opentestsystem/rdw/ingest/repository/RdwImportRepositoryImpl.java]: Options I see (in a semi-random order):
1. write SQL like this 
```sql
SELECT * FROM import
  WHERE
         (:content is null or conent = :content)
    AND  (:status is null or status=:status)
    AND  (:batch is null or batch=:batch)
```

2. Introduce something similar to what we had at SRI with 'optional'. We may need it for the Paging/Sorting anyway.

3. Keep it as is. Admit I like it the least. Defeats the whole idea of sql.yml file.

Comment (malaffoon) [inline exam-service/src/main/java/org/opentestsystem/rdw/ingest/repository/RdwImportRepositoryImpl.java]: I'll experiment with #1, thanks.

**Pull request merged**